src/ambianic/webapp/server/samples.py

```diff
@@ -144,36 +144,60 @@ def get_timeline(before_datetime=None, page=1, data_dir=None):
                         ' Error: %s', before_datetime, str(e))
     page_start_position = (page-1)*page_size
     page_end_position = page_start_position + page_size
+
     if not parsed_datetime:
         log.debug('Fetching most recent saved samples')
     log.debug('Fetching samples page %d. Page size %d. '
               'Sample index range [%d:%d]. ',
               page, page_size, page_start_position, page_end_position)
-    p = Path(data_dir) / 'timeline-event-log.yaml'
-    log.debug('Timeline path: %s', p.resolve())
-    with p.open() as pf:
-        timeline_events = yaml.safe_load(pf)
-    # latest_events_first = timeline_events.reverse()
-    log.debug('Fetched timeline file into struct of type %r with %d events: ',
-              type(timeline_events),
-              len(timeline_events))
-    # events are appended to the file as they arrive
-    # we need to read in reverse order to get the latest one first
-    timeline_slice = \
-        timeline_events[-1*page_start_position - 1:
-                        -1*page_end_position - 1:
-                        -1]
-    # files = sorted(files, key=os.path.getmtime, reverse=True)
-    # for json_file in files[page_start_position:page_end_position]:
-    # if
-    #    with open(json_file) as f:
-    #        sample = json.load(f)
-    #        sample['id'] = uuid.uuid4().hex
-    #        sample['file'] = str(json_file)
-    #        samples.append(sample)
-    # lines = map(str, files)
-    # log.debug('File names follow:\n %s', "\n".join(lines))
-    return timeline_slice
+
+    files = list(Path(data_dir).rglob('timeline-event-log.yaml*'))
```

Comment:
Very nice first cut @muka . Very much appreciate adding tests to keep our coverage high.

As you probably realized, event yaml files are just using the standard file rotation logger, which is supposed to ensure that the latest events are logged in the `timeline-event-log.yaml` file, the next most recent batch is in `timeline-event-log.yaml.1` and so on.

Based on that I think it would be sensible to load event files one at a time as needed using a file suffix increment to find and return the page requested by the API. Would you be interested in implementing such approach?

Loading all event files at ones makes the implementation even more inefficient than it already is. Over time the number of these files can grow substantially. 

I am still monitoring user feedback on performance, but at some point I suspect we will need to move to a time series database that can efficiently scan and return event ranges. I am open to fast forwarding this approach if you'd be more interested going that way.

Reply:
I did it, see the second commit. It should do what you described.

https://github.com/ambianic/ambianic-edge/commit/5a96034aed0ae0eb94e94f7ba68f49ecae6d827c

Reply:
You are fast!! Reviewing now...